### PR TITLE
feat(ci): guard release-publish on existing tag + roll back VERSION on failure

### PR DESCRIPTION
**Pull request by nickolas-dimitrakas**
## Background

Releases are triggered by a change to `VERSION` on `main`/`maintenance/*`. When a `Release – Publish` run fails, `main`'s `VERSION` is already at the new version, so there's no clean way to retrigger without hand-editing `VERSION` — and any such edit risks kicking off *another* release. `main` is protected (PR + approval, no bypass actors), so a workflow can't push a `VERSION` change directly.

## What Has Changed

`.github/workflows/release-publish.yml`:

1. **`preflight` tag-existence guard (new first job).** Reads `VERSION` and checks `gh release view`. If that version already has a GitHub release, it sets `should-release=false` and every downstream job is skipped. This makes any `VERSION` change to an already-released version a safe no-op — a rollback (or a re-run) can never re-publish an existing version.
2. **`build-and-release` and `publish-cocoapods` are gated** on `needs.preflight.outputs.should-release == 'true'`.
3. **`rollback-on-failure` (new job).** If `build-and-release` fails, it opens a PR reverting `VERSION` to the previously-released value (derived from `github.event.before:VERSION`). A human merges it (branch protection); the guard guarantees merging it won't start a new release. Uses the same `peter-evans/create-pull-request` action already used by `Release – Draft`.

Pattern mirrors `ROKT/rokt-sdk-ios`'s gating `detect` job, gating on "not already released."

## Screenshots/Video

- N/A — CI/release-automation change.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation. 
- [ ] I have added tests that prove my fix is effective or that my feature works. 
- [x] I have tested this locally. 

## Behavior / edge cases

- **Retrigger a flaky failure:** just "Re-run failed jobs" — the guard sees no tag (pre-tag failure) and proceeds; no `VERSION` change needed.
- **Merging the rollback PR:** pushes the previous (already-tagged) version → guard stops → no re-release.
- **Failure after the release/tag was created (e.g. only CocoaPods failed):** `build-and-release` succeeded, so rollback does *not* fire; re-running is gated off by the guard. Retry CocoaPods separately.
- **Test failures** are inside `build-and-release`, so they also open a rollback PR — close it and fix-forward if you'd rather.

## Additional Notes

- Stacked on #322 (base = `claude/magical-matsumoto-1a73ce`, which already contains the merged #323). Rebase to `main` once #322 merges.
- Recommend an end-to-end dry-run on a throwaway `maintenance/*` branch before relying on it (see the PR's plan): confirm the guard skips an already-released version and that a forced failure opens the rollback PR. Do not test on `main`.

## Reference Issue (For employees only. Ignore if you are an outside contributor)

- N/A

### CODE REVIEW

File: .github/workflows/release-publish.yml [hunk 1]
```diff
@@ -12,8 +12,44 @@ permissions:
...
+      - name: Checkout code
+        uses: actions/checkout@df4cb1c069e1874edd31b4311f1884172cec0e10 # v6.0.3
+        with:
+          persist-credentials: false
+
+      - name: Get current version
+        id: version-file
+        run: |
+          version_from_file=$(head -n 1 VERSION | tr -d '\r\n ')
+          echo "release-version=$version_from_file" >> "$GITHUB_OUTPUT"
```
[the remark this answers is not in the document]
Comment (nickolas-dimitrakas) [reply .github/workflows/release-publish.yml:43]: Fixed in `7ed01d5`: the `preflight` step now validates the VERSION value against a strict semver regex (`^[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z.-]+)?$`) and fails the run on anything else, before writing it to `GITHUB_OUTPUT`. Because every downstream job `needs: preflight`, this gates the whole workflow, so the release/tag steps are covered too.

File: .github/workflows/release-publish.yml [hunk 2]
```diff
@@ -84,3 +121,55 @@ jobs:
...
+          fi
+          echo "previous-version=$prev" >> "$GITHUB_OUTPUT"
+
+      - name: Roll back VERSION file
+        env:
+          PREV: ${{ steps.prev.outputs.previous-version }}
+        run: printf '%s\n' "$PREV" > VERSION
+
+      - name: Open rollback pull request
+        uses: peter-evans/create-pull-request@5f6978faf089d4d20b00c7766989d076bb2fc7f1 # v8.1.1
```
[the remark this answers is not in the document]
Comment (nickolas-dimitrakas) [reply .github/workflows/release-publish.yml:169]: False positive — the action is already pinned to a full commit SHA, not a tag. Verified that `peter-evans/create-pull-request@v8.1.1` resolves to commit `5f6978faf089d4d20b00c7766989d076bb2fc7f1`, which is exactly the pinned SHA (and the `# v8.1.1` comment is accurate). All other `uses:` in this workflow are SHA-pinned as well. No change needed here.
